argutils/viz.py

```diff
@@ -58,6 +68,33 @@ def draw(ts, ax, use_ranked_times=None, tweak_x=None, arrows=False):
     )
 
 
+def label_nodes(ts, labels=None, metadata_name="name"):
+    """
+    Adds a metadata item (by default "name") to each node, whose value
+    is given by the labels dictionary passed in. If labels is None (default),
+    use the dictionary {0: 'A', 1: 'B', 2: 'C', ... 26: 'Z'}.
+    Any nodes without a corresponding key in the labels dictionary will
+    simply have their metadata value set to their node id.
+    
+    Note that this means that if no labels are given, nodes 26 onwards
+    will be labelled with numbers rather than ascii uppercase letters.
+    """
+    if labels is None:
+        labels = {i: lab for i, lab in enumerate(string.ascii_uppercase)}
+    tables = ts.dump_tables()
+    try:
+        if tables.nodes.metadata_schema.asdict()["codec"] != "json":
```

Comment:
Probably not worth bothering with this - it'll probably end up being as fragile as just going ahead and assuming the codec is JSON.

Reply:
Should I wrap the metadata setting code in a try: except block. I was worried that it would be a confusing error if run on a ts without json metadata.